tailscale/resource_logstream_configuration_test.go

```diff
@@ -0,0 +1,131 @@
+			var selectedConfig *tsclient.LogstreamConfiguration
+			logstreamConfigurationLogTypeConfig, err := client.Logging().LogstreamConfiguration(context.Background(), tsclient.LogTypeConfig)
+			if expectedLogType == tsclient.LogTypeConfig {
+				if err != nil {
+					return err
+				} else {
+					selectedConfig = logstreamConfigurationLogTypeConfig
+				}
+			} else if expectedLogType == tsclient.LogTypeNetwork && err == nil {
+				return fmt.Errorf("expected no configuration logstream configuration but got %+v", logstreamConfigurationLogTypeConfig)
+			}
+
+			logstreamConfigurationLogTypeNetwork, err := client.Logging().LogstreamConfiguration(context.Background(), tsclient.LogTypeNetwork)
+			if expectedLogType == tsclient.LogTypeNetwork {
+				if err != nil {
+					return err
+				} else {
+					selectedConfig = logstreamConfigurationLogTypeNetwork
+				}
+			} else if expectedLogType == tsclient.LogTypeNetwork && err == nil {
+				return fmt.Errorf("expected no network logstream configuration but got %+v", logstreamConfigurationLogTypeNetwork)
+			}
+
+			if selectedConfig.LogType != expectedLogType {
+				return fmt.Errorf("bad logstream_configuration.log_type: %s", selectedConfig.LogType)
+			}
+			if selectedConfig.DestinationType != expectedDestinationType {
+				return fmt.Errorf("bad logstream_configuration.destination_type: %s", selectedConfig.DestinationType)
+			}
+			if selectedConfig.URL != expectedURL {
+				return fmt.Errorf("bad logstream_configuration.url: %s", selectedConfig.URL)
+			}
+			return nil
+		}
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories(t),
+		CheckDestroy: checkResourceDestroyed(resourceName, func(client *tsclient.Client, rs *terraform.ResourceState) error {
+			_, err := client.Logging().LogstreamConfiguration(context.Background(), tsclient.LogType(rs.Primary.ID))
+			if err == nil {
+				return fmt.Errorf("logstream configuration %q still exists on server", resourceName)
+			}
+			return nil
+		}),
+		Steps: []resource.TestStep{
+			{
+				Config: testLogStreamConfiguration,
+				Check: resource.ComposeTestCheckFunc(
+					checkResourceRemoteProperties(
+						resourceName,
+						checkProperties(tsclient.LogTypeConfig, tsclient.LogstreamPantherEndpoint, "https://example.com"),
+					),
+					resource.TestCheckResourceAttr(resourceName, "log_type", "configuration"),
+					resource.TestCheckResourceAttr(resourceName, "destination_type", "panther"),
+					resource.TestCheckResourceAttr(resourceName, "url", "https://example.com"),
+					resource.TestCheckResourceAttr(resourceName, "token", "some-token"),
+				),
+			},
+			{
+				Config: testLogstreamConfigurationUpdateSameLogtype,
+				Check: resource.ComposeTestCheckFunc(
+					checkResourceRemoteProperties(
+						resourceName,
+						checkProperties(tsclient.LogTypeConfig, tsclient.LogstreamCriblEndpoint, "https://example-other.com"),
```

Comment:
Nit - if you want to stick with `example.com` as the domain, you can just add paths, e.g. `https://example.com/path1`